### PR TITLE
MultiValidationBehavior

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###
MultiValidationBehavior allows performing multiple specific validations at the same.
If any validator has IsValid=False, it will cause IsValid=False for the MultiValidationBehavior as well.
Also, it combines "Errors" (attached property) from all "failed" validators. It will allow users to receive detailed info about failed validation.

Example:
```csharp
<Entry Placeholder="Number">
    <Entry.Behaviors>
        <behaviors:MultiValidationBehavior InvalidStyle="{StaticResource InvalidEntryStyle}">
            <behaviors:NumericValidationBehavior behaviors:MultiValidationBehavior.Error="NaN" />
            <behaviors:NumericValidationBehavior MinimumValue="-10"
                                                 behaviors:MultiValidationBehavior.Error="Min: -10" />
            <behaviors:NumericValidationBehavior MaximumValue="5"
                                                 behaviors:MultiValidationBehavior.Error="Max: 5" />
        </behaviors:MultiValidationBehavior>
    </Entry.Behaviors>
</Entry>
```

### Bugs Fixed ###

- Related to issue #204 

### API Changes ###

List all API changes here (or just put None), example:

Added: 
- **MultiValidationBehavior** class

### Behavioral Changes ###
None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of main at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

**Review by sthewissen — APPROVED** [review]
I have left a comment on the original spec, other than that, great.

**Comment by AndreiMisiukevich** [comment]
Any concerns @jfversluis @Char0394 ? :)

File: XamarinCommunityToolkit/Behaviors/Validators/MultiValidationBehavior.shared.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			BindableProperty.Create(nameof(Errors), typeof(List<object>), typeof(MultiValidationBehavior), null, BindingMode.OneWayToSource);
+
+		public static readonly BindableProperty ErrorProperty =
+			BindableProperty.CreateAttached(nameof(GetError), typeof(object), typeof(MultiValidationBehavior), null);
+
+		readonly ObservableCollection<ValidationBehavior> children
+			= new ObservableCollection<ValidationBehavior>();
+
+		public MultiValidationBehavior()
+			=> children.CollectionChanged += OnChildrenCollectionChanged;
```
Comment (jfversluis) [inline XamarinCommunityToolkit/Behaviors/Validators/MultiValidationBehavior.shared.cs:22]: Should this also be unhooked somewhere?

File: XamarinCommunityToolkitSample/ViewModels/Behaviors/BehaviorsGalleryViewModel.cs [hunk 2]
```diff
@@ -10,34 +11,39 @@ public class BehaviorsGalleryViewModel : BaseViewModel
...
 			new SectionModel(
 				typeof(MaskedBehaviorPage),
-				"MaskedBehavior",
+				nameof(MaskedBehavior),
 				AppResources.MaskedShortDescription
 			),
 			new SectionModel(
 				typeof(UriValidationBehaviorPage),
-				"UriValidationBehavior",
+				nameof(UriValidationBehavior),
```
Comment (jfversluis) [inline XamarinCommunityToolkitSample/ViewModels/Behaviors/BehaviorsGalleryViewModel.cs:39]: Great stuff!

**Review by jfversluis — COMMENTED** [review]
Just one small thing. I think we need to unhook that on the detached from to prevent memory leaks?

File: XamarinCommunityToolkit/Behaviors/Validators/MultiValidationBehavior.shared.cs (hunk 1 above)
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkit/Behaviors/Validators/MultiValidationBehavior.shared.cs:22]: I thought .NET garbage collector should know how to detect and resolve cycle references.
Was I wrong?

Comment (jfversluis) [reply XamarinCommunityToolkit/Behaviors/Validators/MultiValidationBehavior.shared.cs:22]: Hm yeah since this lives in the shared bit that should be true.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**